packages/core/utils/lib/traverse-entity.js

```diff
@@ -11,9 +47,13 @@ const traverseEntity = async (visitor, options, entity) => {
   }
 
   // Don't mutate the original entity object
-  const copy = cloneDeep(entity);
+  // only clone at 1st level as the next level will get clone when traversed
+  const copy = clone(entity);
```

Comment:
we could be using a normal object spread here no?
```suggestion
  const copy = {...entity};
```

Seems to be much faster https://www.measurethat.net/Benchmarks/Show/4646/0/object-cloning-with-lodash-clone-vs-es6-object-spread-v
![image](https://user-images.githubusercontent.com/20578351/226665410-77b9c84f-3042-4e31-81db-73e9f928081f.png)

Reply:
the spread operator does clone dates too no?
Well I just tried and it does not make a lot of difference in terms of performance to add this change 🤷

Reply:
I'll just merge as is then we can checj this later. I think spreading only copies scalar values :)

Reply:
FWIW, spreading does in fact copy objects :)

const date = { foo: new Date() }
{foo: Fri Apr 14 2023 08:51:25 GMT-0700}

const newDate = {...date}
{foo: Fri Apr 14 2023 08:51:25 GMT-0700}